src/data/utils.ts

```diff
@@ -0,0 +1,33 @@
+import { QueryBuilderOptions, QueryType } from "types/queryBuilder"
+
+/**
+ * Returns true if the builder options contain enough information to start showing a query
+ */
+export const isBuilderOptionsRunnable = (builderOptions: QueryBuilderOptions): boolean => {
+  return (
+    (builderOptions.columns?.length || 0) > 0 ||
+    (builderOptions.filters?.length || 0) > 0 ||
+    (builderOptions.orderBy?.length || 0) > 0 ||
+    (builderOptions.aggregates?.length || 0) > 0 ||
+    (builderOptions.groupBy?.length || 0) > 0
+  );
+};
+
+/**
+ * Converts QueryType to Grafana format
+ * src: https://github.com/grafana/sqlds/blob/main/query.go#L20
+ */
+export const mapQueryTypeToGrafanaFormat = (t: QueryType): number => {
+  switch (t) {
+    case QueryType.Table:
+      return 1;
+    case QueryType.Logs:
+      return 2;
+    case QueryType.TimeSeries:
+      return 0;
+    case QueryType.Traces:
+      return 3;
+    default:
+      return 1 << 8; // an unused u32, defaults to timeseries/graph on plugin backend.
```

Comment:
[Yes](https://github.com/grafana/sqlds/blob/main/query.go#L22)

However, there is another place in the backend where `PreferredVisualisationType` will default to TimeSeries if `format` isn't provided (or in this case, if the number doesn't match). I wanted to prioritize the backend's logic in case the auto data/panel type detection becomes more advanced in the future.

I also recall trying a simple `-1` but it wanted an unsigned number so thats why I went with an unusably large value. 

v3 used to use `4` to achieve this same effect, but that would cause an issue if they ever added another visualization type